distributor/http.go

```diff
@@ -0,0 +1,56 @@
+package distributor
+
+import (
+	"net/http"
+
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/prometheus/storage/remote"
+
+	"github.com/weaveworks/cortex/util"
+)
+
+// PushHandler is a http.Handler which accepts WriteRequests.
+func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
+	var req remote.WriteRequest
+	ctx, abort := util.ParseProtoRequest(w, r, &req, true)
+	if abort {
+		return
+	}
+
+	_, err := d.Push(ctx, &req)
+	if err != nil {
+		switch e := err.(type) {
+		case IngesterError:
```

Comment:
AIUI, this is only going to be relevant if we're also using the HTTP ingester client, and that with gRPC, we will no longer be able to distinguish between user errors and server errors. Is there a way of addressing this? It's a bit of a regression.